crates/common_utils/src/ext_traits.rs

```diff
@@ -434,3 +435,24 @@ impl XmlExt for &str {
         de::from_str(self)
     }
 }
+
+/// Extension trait for logging the error of a Result and then discarding the error
+pub trait ResultExtLog<T> {
+    ///
+    /// Log error if it is of `Err` type else return the `Ok` value wrapped in `Some`
+    ///
+    fn log_err_and_ok(self, key: &str) -> Option<T>;
+}
+
+impl<T, E> ResultExtLog<T> for Result<T, E>
+where
+    E: std::fmt::Display,
```

Comment:
We could use , but since most of the error types have display impl and usually display impl is logged , so added this. For errors with no Display impl, direct `map_err` can be used.